### PR TITLE
Add scalar function current_time to return string formatted time equi…

### PR DESCRIPTION
**Pull request by marregui**
Missing documentation and unreleased notes with breaking changes.

### CODE REVIEW

File: sql/src/main/java/io/crate/expression/scalar/timestamp/CurrentTimeFunction.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        DataTypes.STRING,
+        FunctionInfo.Type.SCALAR,
+        Collections.emptySet());
+
+    public static void register(ScalarFunctionModule module) {
+        module.register(
+            Signature.scalar(
+                NAME,
+                parseTypeSignature("integer"),
+                parseTypeSignature("string")
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/expression/scalar/timestamp/CurrentTimeFunction.java:58]: This function should return time with time zone, not text

Comment (marregui) [reply sql/src/main/java/io/crate/expression/scalar/timestamp/CurrentTimeFunction.java:58]: Thank you for that!, I now understand. Time just gives time! the time portion of the timestamp only, thank you, will amend.

**Comment by marregui** [comment]
retest this please

Looks like the process got stuck?

File: docs/general/builtins/scalar.rst [hunk 2]
```diff
@@ -816,6 +816,22 @@ are given.
...
+
+``CURRENT_TIME``
+----------------
+
+The ``CURRENT_TIME`` expression returns the time portion of ``CURRENT_TIMESTAMP``
+in absolute milliseconds. The same time value is returned for every invocation of
+a single statement, and will match the time portion of invocations to
+``CURRENT_TIMESTAMP`` in the same statement.
+
+synopsis::
```
Comment (mechanomi) [inline docs/general/builtins/scalar.rst:829]:
```suggestion
Synopsis::
```

File: docs/general/builtins/scalar.rst [hunk 3]
```diff
@@ -816,6 +816,22 @@ are given.
...
+
+    CURRENT_TIME [ ( precision ) ]
+
+Where precision is defined in :ref:`current_timestamp`.
```
Comment (mechanomi) [inline docs/general/builtins/scalar.rst:833]:
```suggestion
Where ``precision`` is defined in :ref:`current_timestamp`.
```

**Comment by kovrus** [comment]
Is it implemented just for the sake of pg compatibility? If it is like that, why don't we just add the alias for the `current_timestamp` function and document that the date part should be ignored (it is essentially what happens in this implementation). Anyways the `current_time` function does not return the correct data type now (such as we are missing it).

**Comment by marregui** [comment]
Just to clarify the functionality, in postgres:

```
select current_timestamp, round(extract(epoch from current_timestamp) * 1000), current_time, round(extract(epoch from current_time) * 1000);
       current_timestamp       |     round     |    current_time    |  round
-------------------------------+---------------+--------------------+----------
 2020-04-15 11:05:41.125711+02 | 1586941541126 | 11:05:41.125711+02 | 32741126
(1 row)
```

obviously the `32741126` when interpreted as a timestamp it would be UTC from 1st Jan 1970.
I suppose if the date portion of a timestamp is irrelevant for a given day only time series, it may be more efficient to store an integer column.

**Comment by kovrus** [comment]
@marregui ah thanks, got a bit confused after some mentions of time string formating and tests. Thanks for the clarification!

**Comment by mfussenegger** [comment]
> I have a question regarding this function. Is it implemented just for the sake of pg compatibility? If it is like that, why don't we just add the alias for the `current_timestamp` function and document that the date part should be ignored, such as anyways it has to be ignored regardless whether it is 1970.. or any other date (it is essentially what happens in this implementation). Anyways the `current_time` function does not return the correct data type now (such as we are missing it), so I am just wondering what was the idea here.

We should add a time type as part of this.

**Comment by marregui** [comment]
Time with and without timezone I suppose

**Comment by marregui** [comment]
will create a new PR salvaging code from this one, but branching of  `ma/current_time_new_type`

**Pull request closed without merging**